### PR TITLE
Subot configs

### PR DESCRIPTION
**Pull request by tiago369**
#### **Overview**

- \<Improves nerian camera functionalities.\>
- \<Separate camera info, add param to change tf of topics, and optimize the code.\>

#### **Use the following settings in the overrides file for docker testing**

- Use these settings to compile and run the module using `autoproj`.
- The contents of the files manifest and `overrides.yaml` must be changed according to the `templates` below.
- Additional packages may be needed to test this module.

  e.g.,

- _overrides.yaml_

  ```yaml
    overrides:
      - nerian_stereo_ros2:
        github: Brazilian-Institute-of-Robotics/nerian_stereo_ros2
        branch: subot-configs
  ```

- _manifest_

  ```yaml
    package_sets:
      - github: Brazilian-Institute-of-Robotics/bir.subot-package_set
        private: true
    layout:
      - nerian_stereo_ros2
  ```

#### **What was added/changed in this update**

- Camera info splitter node
- Add params to change frame ids of topics
- Correct rclcpp::time calculation

#### **Depends On:**

- N/A

#### **Related Issues:**

- N/A

#### **Notes:**

- N/A

### CODE REVIEW

File: src/camera_info_splitter.cpp [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+            camera_right_info_->publish(msg->right_info);            
+        }
+};
+
+int main(int argc, char* argv[]){
+    rclcpp::init(argc, argv);
+    rclcpp::spin(std::make_shared<CameraInfoSplitter>());
+    rclcpp::shutdown();
+    return 0;
+}
```
Comment (leonejesus) [inline src/camera_info_splitter.cpp]: Please add an EoF line.

File: src/camera_info_splitter.cpp [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        rclcpp::Publisher<sensor_msgs::msg::CameraInfo>::SharedPtr camera_right_info_;
+
+        void nerian_info_callback(const nerian_stereo::msg::StereoCameraInfo::SharedPtr msg) const
+        {
+            camera_left_info_->publish(msg->left_info);
+            camera_right_info_->publish(msg->right_info);            
+        }
+};
+
+int main(int argc, char* argv[]){
```
Comment (leonejesus) [inline src/camera_info_splitter.cpp]: You should separate this file in a main, .hpp and .cpp

File: src/nerian_stereo_node.cpp [hunk 3]
```diff
@@ -769,8 +777,8 @@ void StereoNode::publishCameraInfo(rclcpp::Time stamp, const ImageSet& imageSet)
         }
     }
 
-    double dt = (stamp - lastCamInfoPublish).seconds();
-    if(dt > 1.0) {
+    double dt = (stamp.seconds() - lastCamInfoPublish.seconds());
+    if(dt > 0.50){
```
Comment (leonejesus) [inline src/nerian_stereo_node.cpp:781]: Could this value be passed in a config file?

File: src/camera_info_splitter.cpp (hunk 2 above)
Comment (tiago369) [reply src/camera_info_splitter.cpp]: Done

File: src/camera_info_splitter.cpp (hunk 1 above)
Comment (tiago369) [reply src/camera_info_splitter.cpp]: Done

File: include/nerian_stereo/camera_info_splitter.hpp [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    private:
+        //Declaring the subscriber variable
+        rclcpp::Subscription<nerian_stereo::msg::StereoCameraInfo>::SharedPtr nerian_camera_info_;
+
+        //Declare the publishers for the camera left and right
+        rclcpp::Publisher<sensor_msgs::msg::CameraInfo>::SharedPtr camera_left_info_;
+        rclcpp::Publisher<sensor_msgs::msg::CameraInfo>::SharedPtr camera_right_info_;
+
+        void nerian_info_callback(const nerian_stereo::msg::StereoCameraInfo::SharedPtr msg) const;
+};
```
Comment (leonejesus) [inline include/nerian_stereo/camera_info_splitter.hpp]: Please add EoF line

File: src/main.cpp [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+#include <string>
+
+#include "nerian_stereo/camera_info_splitter.hpp"
+
+int main(int argc, char* argv[]){
+    rclcpp::init(argc, argv);
+    rclcpp::spin(std::make_shared<CameraInfoSplitter>());
+    rclcpp::shutdown();
+    return 0;
+}
```
Comment (leonejesus) [inline src/main.cpp]: Please add EoF line

**Review by leonejesus — COMMENTED** [review]
Please add the correct documentation in Doxygen for the files you created.
You can see an example here:
https://github.com/Brazilian-Institute-of-Robotics/subot_behavior_tree/blob/main/include/subot_behavior_tree/actions/inspection_client.hpp

Use cpplint on these files too. You probably will see problems with headers order, that should be: Related header, C system headers, C++ standard library headers, other libraries' headers, your project's headers.

Also, some of them are using "..." instead of <...> (even not being a package member).

File: include/nerian_stereo/camera_info_splitter.hpp (hunk 4 above)
Comment (tiago369) [reply include/nerian_stereo/camera_info_splitter.hpp]: Done

File: src/main.cpp (hunk 5 above)
Comment (tiago369) [reply src/main.cpp]: Done

**Comment by tiago369** [comment]
> Please add the correct documentation in Doxygen for the files you created. You can see an example here: https://github.com/Brazilian-Institute-of-Robotics/subot_behavior_tree/blob/main/include/subot_behavior_tree/actions/inspection_client.hpp
> 
> Use cpplint on these files too. You probably will see problems with headers order, that should be: Related header, C system headers, C++ standard library headers, other libraries' headers, your project's headers.
> 
> Also, some of them are using "..." instead of <...> (even not being a package member).

Required changes made

File: src/nerian_stereo_node.cpp (hunk 3 above)
Comment (tiago369) [reply src/nerian_stereo_node.cpp:781]: This data does not need to be changed frequently

**Review by leonejesus — APPROVED** [review]

**Review by argolomb — APPROVED** [review]

**Pull request merged**